specification/web/resource-manager/Microsoft.Web/stable/2022-09-01/StaticSites.json

```diff
@@ -3811,6 +4562,130 @@
         }
       }
     },
+    "StaticSiteDatabaseConnectionConfigurationFileOverview": {
+      "description": "A database connection configuration file",
+      "type": "object",
+      "properties": {
+        "fileName": {
+          "description": "The name of the configuration file.",
+          "type": "string",
+          "readOnly": true
+        },
+        "contents": {
+          "description": "The Base64 encoding of the file contents.",
+          "type": "string",
+          "readOnly": true
+        },
+        "type": {
+          "description": "The type of configuration file.",
+          "type": "string",
+          "readOnly": true
+        }
+      },
+      "x-ms-client-flatten": true
+    },
+    "DatabaseConnection": {
+      "description": "Static Site Database Connection ARM resource.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "./CommonDefinitions.json#/definitions/ProxyOnlyResource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "description": "DatabaseConnection resource specific properties",
+          "type": "object",
+          "properties": {
+            "resourceId": {
+              "description": "The resource id of the database.",
+              "type": "string"
+            },
+            "connectionIdentity": {
+              "description": "If present, the identity is used in conjunction with ConnectionString to connect to the database. Use of the system-assigned managed identity is indicated with the string 'SystemAssigned', while use of a user-assigned managed identity is indicated with the resource id of the managed identity resource.",
+              "type": "string"
+            },
+            "connectionString": {
+              "description": "The connection string to use to connect to the database.",
+              "type": "string",
+              "x-ms-secret": true
+            },
+            "region": {
+              "description": "The region of the database resource.",
+              "type": "string"
+            },
```

Comment:
Naming: Would it be clearer to call it 'databaseLocation'? Since if you just see 'region' without context, you might assume its the region of the connection (whatever that would mean) or perhaps the static site?

Reply:
Is this field mutable? Or set at create time? Or is it effectively readonly because the value is automatically determined by which database resource id you pass in?

Reply:
The field is indeed mutable.

Regarding the naming for `region` and `resourceId` we specifically went with these rather than prefixing with `database` because we found that redundant within database connection requests/responses (all the properties relate to the database). Let me know if this would be a blocking requirement, as we've already built out a lot of this feature with these names and it would be quite a bit of work to revise and redeploy.

Reply:
@joslinmicrosoft They are just suggestions, as resourceId is in general 'reasonable' and in line with guidelines . I went with these suggestions partly because, although the naming may seem redundant today, it may _not_ seem as redundant or clear in future API versions if you add in more properties that reference more resources.